### ISSUE TITLE
Ranger unresponsive while preview is loading

### ISSUE THREAD

**Issue by applesud** [ranger/ranger#2170]
#### Runtime Environment

- Operating system and version: Debian Buster
- Terminal emulator and version: kitty 0.13.3-1 & xterm 344-1
- Python version: 3.7.3-1 & 2.7.16-1
- Ranger version/commit: 1.9.2-4
- Locale: en_AU.utf8

#### Current Behavior
On moving the cursor over an image with previews enabled, ranger becomes unresponsive and keypresses are not registered until preview has loaded. This happens in both kitty and xterm with both w3m and kitty previews.

#### Expected Behavior
Ranger allows keypresses while loading previews, and cancels loading of preview if cursor is moved off file before loading is complete.

#### Steps to reproduce
1. Enable image previews
2. Move cursor onto an image file (effect is more obvious with larger file sizes)
3. Try to perform an action: for instance, attempt to move cursor off image file.
4. It will not work until image has finished loading

**Comment by toonn** [comment]
Ranger previews are synchronous for the time being. You can cancel them with `^C`. This isn't ideal and it's the reason some previews are commented out by default. You could also toggle image previews off whenever you want to navigate quickly, using `zi`. There's a couple open PRs that will help alleviate this.

**Comment by jedi2610** [comment]
could this be a duplicate of #2089?

**Comment by applesud** [comment]
@jedi2610 No, the image does load and CPU usage is normal. It's just that, because ranger loads images synchronously, ranger waits until the image has finished loading before doing anything else (like responding to keypresses). The effect is quite subtle except on very large images, where it can disrupt normal usage somewhat.
